File: trunk/openrest4js/openrest4js-api/Tag.js
OPENREST_TAG_MODE_INCLUDE = "include";
OPENREST_TAG_MODE_EXCLUDE = "exclude";
	
function TagFromObj(data)
{
    return new Tag(data.id, data.restaurantId, data.title, data.itemIds, data.properties);
}

function Tag(id, restaurantId, title, itemIds, properties) {
    this.id = id;
    this.restaurantId = restaurantId;
    this.title = title;
    this.itemIds = itemIds;
    this.properties = properties;
}

Tag.prototype.toOpenRestObj = function()
{
    var ret = {};

    if (typeof(this.id) != "undefined") ret['id'] = this.id;
    if (typeof(this.restaurantId) != "undefined") ret['restaurantId'] = this.restaurantId;
    if (typeof(this.title) != "undefined") ret['title'] = this.title;
    if (typeof(this.itemIds) != "undefined") ret['itemIds'] = this.itemIds;
    if (typeof(this.properties) != "undefined") ret['properties'] = this.properties;

    return ret;
}

File: openrest4cocoa/openrest4cocoa/openrest4cocoa-api/Tag.m
//
//  Tag.m
//  openrest4cocoa
//
//  Created by Yoav Amit on 5/4/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "Tag.h"


@implementation Tag

@synthesize tagId;
@synthesize restaurantId;
@synthesize title;
@synthesize itemIds;


-(id)init
{
    if ((self = [super init]))
    {
        [self setItemIds:[NSArray array]];
    }
    return self;
}

-(id)initWithDictionary:(NSDictionary*)data
{
    if ((self = [self init]))
    {
        [self setTagId:[data valueForKey:@"id"]];
        [self setRestaurantId:[data valueForKey:@"restaurantId"]];
        [self setTitle:[LocalizedDictionary dictionaryWithDictionary:[data valueForKey:@"title"]]];
        if ([data valueForKey:@"itemIds"] != nil)
        {
            [self setItemIds:[data valueForKey:@"itemIds"]];
        }
    }
    return self;
}

-(NSUInteger)hash
{
    return [tagId hash];
}

-(BOOL)isEqual:(id)object
{
    if (object == self) return TRUE;
    if (object == NULL) return FALSE;
    if (![object isKindOfClass:[self class]]) return FALSE;
    Tag* other = (Tag*)object;

    if (![tagId isEqualToString:other.tagId]) return FALSE;
    if (![restaurantId isEqualToString:other.restaurantId]) return FALSE;
    if (![title isEqual:other.title]) return FALSE;
    if (![itemIds isEqualToArray:other.itemIds]) return FALSE;

    return TRUE;
}

-(NSDictionary*)proxyForJson
{
    NSMutableDictionary* ret = [NSMutableDictionary dictionaryWithCapacity:0];

    if (tagId != nil) {[ret setValue:tagId forKey:@"id"];}
    if (restaurantId != nil) {[ret setValue:restaurantId forKey:@"restaurantId"];}
    if (title != nil) {[ret setValue:title forKey:@"title"];}
    if (itemIds != nil) {[ret setValue:itemIds forKey:@"itemIds"];}
    
    return ret;
}

-(NSString*)description
{
    return [NSString stringWithFormat:@"Tag[%@]: %@ %@ %@", tagId, title, restaurantId, itemIds];
}

-(void)dealloc
{
    [tagId release];
    [restaurantId release];
    [title release];
    [itemIds release];
    [super dealloc];
}

@end
